Flow.Launcher.Core/Plugin/PluginConfig.cs

```diff
@@ -45,8 +45,56 @@ public static List<PluginMetadata> Parse(string[] pluginDirectories)
                     }
                 }
             }
- 
-            return allPluginMetadata;
+
+            (List<PluginMetadata> uniqueList, List<PluginMetadata> duplicateList) = GetUniqueLatestPluginMetadata(allPluginMetadata);
+
+            duplicateList
+                .ForEach(
+                    x => Log.Warn("PluginConfig",
+                                    string.Format("Duplicate plugin name: {0}, id: {1}, version: {2} " +
+                                                    "not loaded due to version not the highest of the duplicates",
+                                                    x.Name, x.ID, x.Version),
+                                                    "GetUniqueLatestPluginMetadata"));
+
+            return uniqueList;
+        }
+
+        internal static (List<PluginMetadata>, List<PluginMetadata>) GetUniqueLatestPluginMetadata(List<PluginMetadata> allPluginMetadata)
+        {
```

Comment:
hey @taooceros can you see a better way of doing this loop?

Reply:
ok I will take a try

Reply:
Hm I think it is good enough. The plugin count will be relatively small, so no need to preoptimize it until we reach a bottleneck for that.